InvalidKeyException when running with export-grade crypto

For Reasons, standard JRE installs do not support sane cryptography by default; you have to install a policy file. This causes the benchmark runs to fail on a stock install with the error below. To make this more usable, we should either

1. change the ART code to use only small key sizes, or
1. add a human-readable error message and instructions for installing the policy file

```
[2018-21-08 15:21:09.886] Illegal key size
java.security.InvalidKeyException: Illegal key size
        at javax.crypto.Cipher.checkCryptoPerm(Cipher.java:1039)
        at javax.crypto.Cipher.implInit(Cipher.java:805)
        at javax.crypto.Cipher.chooseProvider(Cipher.java:864)
        at javax.crypto.Cipher.init(Cipher.java:1396)
        at javax.crypto.Cipher.init(Cipher.java:1327)
        at com.facebook.research.asynchronousratchetingtree.crypto.Crypto.encrypt(Crypto.java:123)
        at com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchet.sendMessage(DHRatchet.java:92)
        at com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchet.sendMessage(DHRatchet.java:23)
        at com.facebook.research.asynchronousratchetingtree.Main.testRun(Main.java:190)
        at com.facebook.research.asynchronousratchetingtree.Main.dhTestRun(Main.java:102)
        at com.facebook.research.asynchronousratchetingtree.Main.main(Main.java:28)
Exception in thread "main" java.lang.RuntimeException
        at com.facebook.research.asynchronousratchetingtree.Utils.except(Utils.java:65)
        at com.facebook.research.asynchronousratchetingtree.crypto.Crypto.encrypt(Crypto.java:138)
        at com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchet.sendMessage(DHRatchet.java:92)
        at com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchet.sendMessage(DHRatchet.java:23)
        at com.facebook.research.asynchronousratchetingtree.Main.testRun(Main.java:190)
        at com.facebook.research.asynchronousratchetingtree.Main.dhTestRun(Main.java:102)
        at com.facebook.research.asynchronousratchetingtree.Main.main(Main.java:28)
make: *** [run] Error 1
```

--- a/AsynchronousRatchetingTree/src/main/java/com/facebook/research/asynchronousratchetingtree/Main.java
+++ b/AsynchronousRatchetingTree/src/main/java/com/facebook/research/asynchronousratchetingtree/Main.java
@@ -16,13 +16,17 @@ import com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchet;
 import com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchetSetupPhase;
 import com.facebook.research.asynchronousratchetingtree.dhratchet.DHRatchetState;
 
+import java.security.NoSuchAlgorithmException;
 import java.security.SecureRandom;
 import java.util.*;
+import javax.crypto.Cipher;
 
 public class Main {
   private static boolean debug = true;
 
   public static void main(String[] args) {
+    checkForUnlimitedStrengthCrypto();
+
     // Run a test run first to warm up the JIT
     artTestRun(8, 8);
     dhTestRun(8, 8);
@@ -90,6 +94,24 @@ public class Main {
       new ARTSetupPhase(),
       new ARTTestImplementation()
     );
+  }
+
+  private static void checkForUnlimitedStrengthCrypto() {
+    boolean hasEnoughCrypto = true;
+    try {
+      if (Cipher.getMaxAllowedKeyLength("AES") < 256) {
+        hasEnoughCrypto = false;
+      }
+    } catch (NoSuchAlgorithmException e) {
+      hasEnoughCrypto = false;
+    }
+    if (!hasEnoughCrypto) {
+      Utils.except("Your JRE does not support unlimited-strength " +
+		   "cryptography and thus cannot run ART. See README.md " +
+		   "for Java Cryptography Extension (JCE) Unlimited " +
+		   "Strength Jurisdiction Policy Files installation " +
+		   "instructions.");
+    }
   }
 
   private static TestResult dhTestRun(int n, int activePeers) {

